### PR TITLE
chore: add M2M RBAC changeset for @logto/core and update changelog

### PR DESCRIPTION
**Pull request by darcyYe**
## Summary

Add M2M RBAC changeset for @logto/core and update changelog for @logto/console.

## Testing

N/A

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] docs

OR

- [x] This PR is not applicable for the checklist

### CODE REVIEW

File: .changeset/early-squids-cheat.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
+"@logto/core": minor
+---
+
+### Minor changes:
+
+Feature: Machine-to-machine (M2M) role-based access control (RBAC)
```
Comment (gao-sun) [inline .changeset/early-squids-cheat.md:7]:
```suggestion
machine-to-machine (M2M) role-based access control (RBAC)

### Summary
```

File: .changeset/early-squids-cheat.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+Following new APIs are added for M2M role management:
+
+- `POST /applications/:appId/roles` assigns role(s) to the M2M application
+- `DELETE /applications/:appId/roles/:roleId` deletes the role from the M2M application
+- `GET /applications/:appId/roles` lists all roles assigned to the M2M application
+
+- `POST /roles/:roleId/applications` assigns the role to multiple M2M applications
+- `DELETE /roles/:roleId/applications/:appId` removes the M2M application assigned to the role
+- `GET /roles/:roleId/applications` lists all M2M applications granted with the role
```
Comment (gao-sun) [inline .changeset/early-squids-cheat.md:23]:
```suggestion
**Applications**
- `POST /applications/:appId/roles` assigns role(s) to the M2M application
- `DELETE /applications/:appId/roles/:roleId` deletes the role from the M2M application
- `GET /applications/:appId/roles` lists all roles assigned to the M2M application

**Roles**
- `POST /roles/:roleId/applications` assigns the role to multiple M2M applications
- `DELETE /roles/:roleId/applications/:appId` removes the M2M application assigned to the role
- `GET /roles/:roleId/applications` lists all M2M applications granted with the role
```

File: .changeset/early-squids-cheat.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Updated following API:
+
+- `POST /roles` to specify the role type (either `user` or `machine-to-machine` role)
+
+- `POST /users/:userId/roles` to prevent assigning M2M roles to end-users
```
Comment (gao-sun) [inline .changeset/early-squids-cheat.md:29]: ditto

**Review by gao-sun — APPROVED** [review]

File: .changeset/early-squids-cheat.md (hunk 3 above)
Comment (darcyYe) [reply .changeset/early-squids-cheat.md]: Updated.

**Pull request merged**